packages/core/src/utils/connectors/factories.ts

```diff
@@ -7,17 +7,36 @@ import { getConnectorPackagesFromDirectory } from '@logto/cli/lib/utils.js';
 import { findPackage } from '@logto/shared';
 import chalk from 'chalk';
 
+import RequestError from '#src/errors/RequestError/index.js';
 import type { ConnectorFactory } from '#src/utils/connectors/types.js';
 
 import { notImplemented } from './consts.js';
 import { parseMetadata, validateConnectorModule } from './index.js';
 import { loadConnector } from './loader.js';
 
+const checkDuplicateConnectorFactoriesId = (connectorFactories: ConnectorFactory[]) => {
+  const connectorFactoriesId = connectorFactories.map(({ metadata }) => metadata.id);
+  const connectorFactoriesIdSet = new Set(connectorFactoriesId);
```

Comment:
```suggestion
  const connectorFactoriesId = deduplicate(connectorFactories.map(({ metadata }) => metadata.id));
```

Reply:
Refactored. We should better keep both `connectorFactoryIds` and `deduplicatedConnectorFactoryIds` because we are trying to give the details on duplicated factory ids for better debugging experience.